### ISSUE TITLE
Changes from version 1.2.0 and 1.2.3 break declaration generation

### ISSUE THREAD

**Issue by alejandroclaro** [wessberg/rollup-plugin-ts#67]
Changes from version 1.2.0 and 1.2.3 break declaration generation. For example:

``` typescript
// module.ts
export default class DefaultClass {
  private static readonly constant = 0;

  public static staticMethod(): void {
  	console.log(DefaultClass.constant)
  }
}
```

in version 1.1.83, and previous, produces a correct declaration file (.d.ts) with this rollup configuration:

``` javascript
const Path = require('path');

const alias = require('@rollup/plugin-alias');
const builtins = require('builtin-modules');
const commonjs = require('@rollup/plugin-commonjs');
const filesize = require('rollup-plugin-filesize');
const json = require('@rollup/plugin-json');
const resolve = require('@rollup/plugin-node-resolve');
const typescript = require('@wessberg/rollup-plugin-ts');

const config = {
  treeshake: false,
  plugins: [
    alias({
      'resolve': [ '.js', '.ts', '' ],
      'entries': [
        { 'find': '@src', 'replacement': Path.join(__dirname, 'src') },
      ]
    }),
    resolve(),
    commonjs(),
    json({ 'compact': true }),
    filesize(),
    typescript({
      'tsconfig': (resolvedConfig) => {
        const config = Object.assign({}, resolvedConfig);
        config.declarationDir = Path.join(__dirname, 'dist', 'types');
        return config;
      }
    })
  ],
  external: [
  	...builtins
  ],
  input: './src/module.ts',
  output: [
    { file: 'dist/rollup-plugin-ts-bug.cjs.js', format: 'cjs', exports: 'named', sourcemap: true } 
  ]
}

export default config;
````

and this tsconfig.json:

``` json
{
  "exclude": [ "node_modules", "dist" ],
  "compilerOptions": {
    "target": "es2018",
    "module": "esnext",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "composite": true,
    "declaration": true,
    "declarationMap": true,
    "experimentalDecorators": true,
    "incremental": true,
    "inlineSources": false,
    "inlineSourceMap": false,
    "listEmittedFiles": true,
    "noEmitOnError": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "removeComments": true,
    "sourceMap": true,
    "strict": true,
    "outDir": "./dist",
    "rootDirs": [ "./src" ],
    "baseUrl": ".",
    "paths": {
      "@src/*": [ "src/*" ],
  	}
  },
  "include": [
    "./src/**/*"
  ],
  "exclude": [
    "./node_modules",
    "./dist"
  ]
}
```

However, with version 1.2.3, rollup/tsc emits the follow error:

``` markdown
[!] (plugin Typescript) TS6304: Composite projects may not disable declaration emit.
```

and version 1.2.0, produces a declaration file additional information at the end of the file that produces an error later when another package needs the declaration file:

``` json
{
  "bundle": {
    "commonSourceDirectory": "../..",
    "sourceFiles": [
      "../../src/module.ts"
    ],
    "dts": {
      "sections": [
        {
          "pos": 0,
          "end": 138,
          "kind": "text"
        }
      ]
    }
  },
  "version": "3.7.4"
}
```

Versions:

- rollup: 1.28.0
- typescript: 3.7.4
- @rollup/plugin-commonjs: 11.0.1
- @rollup/plugin-node-resolve: 6.1.0

**Comment by Marsup** [comment]
I have the same bug too, it seems to be coming from the incremental build. The output files [here](https://github.com/wessberg/rollup-plugin-ts/blob/master/src/declaration/bundle-declarations-for-chunk.ts#L14) include the cache itself for some reason. Disabling the incremental feature fixes the output.

**Comment by wessberg** [comment]
Hey there. This has been fixed in [v1.2.4](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v1.2.4). The old output you've seen in which the contents of the `.tsbuildinfo` file was inlined directly in the declaration file was a bug and due to the fact that incremental compilation via the `incremental` and `composite` properties simply wasn't supported by `rollup-plugin-ts`. Thanks to your issue, I've gone ahead and added support for it!

**Comment by alejandroclaro** [comment]
@wessberg thanks for such quick fix!
